src/bp/ui-studio/src/web/views/FlowBuilder/diagram/manager.ts

```diff
@@ -119,8 +119,24 @@ export class DiagramManager {
     this.getLinksRequiringUpdate()
   }
 
-  shouldHighlightNode(nodeName): boolean {
-    return this.highlightedNodeNames && !!this.highlightedNodeNames.find(x => nodeName.includes(x))
+  shouldHighlightNode(node: NodeView): boolean {
+    if (_.isEmpty(this.highlightedNodeNames)) {
+      return false
+    }
+
+    return this.highlightedNodeNames
+      .map(q => {
+        // quick and dirty way to filter on node content
+        const candidates = [
+          node.name,
+          node.id,
+          JSON.stringify(node.contents || {}),
+          JSON.stringify(node.prompt?.params.question ?? {}),
+          JSON.stringify(node.conditions || {})
+        ]
```

Comment:
What about just stringifying the whole node instead ? You'll even pick new additions without having to change anything.   Remove keys & common objects and we're good to go!

Reply:
I thought about that too, but we might get weird behaviours if for example node.id and node.name make a string together by combining them, but not when separated. It could create weird edgecases

Reply:
@allardy doesn't work, we end up with loads of keys as candidate for search. Most of nodes get highlighted in the first 3 or 4 key strokes and it's quite annoying